Makefile

```diff
@@ -178,3 +178,22 @@ build-webhook:
 .PHONY: docker-build-webhook
 docker-build-webhook:
 	docker build -f cmd/admission-webhook-server/Dockerfile -t $(ADMISSION_WEBHOOK_SERVER) .
+
+.PHONY: check-working-tree-clean
+check-working-tree-clean:
+	@test -z "$$(git status --porcelain)" || (echo "Commit all changes before running this target"; exit 1)
+
+.PHONY: check-update-assets
+check-update-assets: check-working-tree-clean update-assets
+	@test -z "$$(git status --porcelain)" || (echo "Please run make update-assets and commit the changes.";git --no-pager diff; exit 1)
+
+.PHONY: check-vendor
+check-vendor: check-working-tree-clean vendor
+	@test -z "$$(git status --porcelain)" || (echo "Please run make vendor and commit the changes.";git --no-pager diff; exit 1)
```

Comment:
We've used `git status` instead of `git --no-pager diff` in vendor and docs checks but I guess using `git diff` doesn't hurt and shows more information :+1:

Reply:
Ah, good catch. It should use `git status`, as `git diff` will print A LOT if you forget to vendor dependencies :smile:

Reply:
Yes, I guess that's a big drawback for `git diff`. Let's use `git status` then.